gammapy/utils/random/inverse_cdf.py

```diff
@@ -0,0 +1,79 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+import numpy as np
+from .utils import get_random_state
+
+
+class InverseCDFSampler:
+    """Inverse CDF sampler.
+       
+   It determines a set of random numbers and calculate the cumulative 
+   distribution function.
+   
+   Parameters
+   ----------
+   pdf : `~gammapy.maps.Map`
+        Map of the predicted source counts.
+   axis : integer
+        Axis along which sampling the indexes.
+   random_state : integer
+        Take a `numpy.random.RandomState` instance.
+   """
+
+    def __init__(self, pdf, axis=None, random_state=0):
+        self.random_state = get_random_state(random_state)
+        self.axis = axis
+
+        if axis is not None:
+            self.cdf = np.cumsum(pdf, axis=self.axis)
+            self.cdf /= self.cdf[:, [-1]]
+        else:
+            self.pdf_shape = pdf.shape
+
+            pdf = pdf.ravel() / pdf.sum()
+            self.sortindex = np.argsort(pdf, axis=None)
+
+            self.pdf = pdf[self.sortindex]
+            self.cdf = np.cumsum(self.pdf)
+
+    def sample_axis(self):
+        """Sample along a given axis.
+
+        Returns
+        -------
+        index : tuple of `~numpy.ndarray`
+            Coordinates of the drawn sample.
+
+        """
+        choice = self.random_state.uniform(high=1, size=len(self.cdf))
+
+        # find the indices corresponding to this point on the CDF
+        index = np.argmin(np.abs(choice.reshape(-1, 1) - self.cdf), axis=self.axis)
+
+        return index + self.random_state.uniform(low=-0.5, high=0.5, size=len(self.cdf))
+
+    def sample(self, size):
+        """Draw sample from the given PDF.
+
+        Parameters
+        ----------
+        size : int
+            Number of samples to draw.
+
+        Returns
+        -------
+        index : tuple of `~numpy.ndarray`
```

Comment:
Is `index` really a tuple. Looking at the code, I think it's a `numpy.ndarray`?

Reply:
Okay. We'll modify the string.